src/consumer.rs

```diff
@@ -1550,4 +1630,95 @@ mod tests {
 
         rt.block_on(f);
     }
+
+    #[test]
+    #[cfg(feature = "tokio-runtime")]
+    fn dead_letter_queue() {
+        use rand::{distributions::Alphanumeric, Rng};
+        let _ = log::set_logger(&TEST_LOGGER);
+        let _ = log::set_max_level(LevelFilter::Debug);
+        let addr = "pulsar://127.0.0.1:6650";
+        let mut rt = Runtime::new().unwrap();
+
+        let topic = "dead_letter_queue_test";
```

Comment:
I've started making these topics include a random u16 in order to remove stale data from previous runs of the test. It might be worth doing that here, something like
```
let test_id: u16 = rand::random();
let topic = format!("dead_letter_queue_test_{}, test_id);
...
let dead_letter_topic = format!("{}_dlq", &topic);
```